### ISSUE TITLE
Extend configuration for SMTP providers

### ISSUE THREAD

**Issue by stebenz** [zitadel/zitadel#8042]
## Problem + Context  
Developers struggle to integrate their preferred SMTP provider with ZITADEL Cloud. Today, the system only supports **basic username/password auth**, which works for some providers (SendGrid, Postmark, SES) but **fails completely for Gmail and Microsoft Exchange Online**, where Basic Auth is disabled.  

Industry shifts make this more urgent:  
- **Microsoft Exchange Online** has permanently removed Basic Auth and now requires **Modern Auth (OAuth2)**.  
- **Gmail SMTP** also enforces OAuth2 for most accounts.  
- Even providers that still accept Basic Auth (SendGrid, SES, Postmark) often have quirks, such as requiring an API key or a specific hostname during the connection handshake.  

On top of that, when no SMTP provider is configured, ZITADEL returns **unclear, untranslated error messages** that create user frustration and support tickets.  

---

## Goals  
- **Primary Goal:** Allow developers to use their existing SMTP provider with ZITADEL Cloud, regardless of whether it uses Basic Auth or Modern Auth.  
- **Secondary Goals:**  
  - Make setup easier and safer with helpful defaults and recovery paths.  
  - Provide clear, localized error messages when something goes wrong.  

---

## TL;DR (User Journey)  
- Developers can connect **any major SMTP provider**: SendGrid, SES, Postmark, Gmail, or Microsoft Exchange Online.  
- They can **choose the right authentication method** for their provider (Basic Auth or OAuth2).  
- If a provider requires **special credentials** (e.g., SendGrid API key, Gmail OAuth token), they can configure that directly.  
- If setup fails, they see a **clear error message** instead of cryptic failures.  
- If their provider rejects the default connection introduction, they can **set a custom hostname for ZITADEL to use when identifying itself**.  

---

## Provider ↔ Auth Method Matrix  

| Provider | PLAIN/LOGIN | API key (as password)* | CRAM-MD5 | XOAUTH2 / OAuth2 | Notes |
|---|---|---|---|---|---|
| **SendGrid** | ✅ | ✅ | ❌ | ❌ | Username = `apikey`, password = API key |
| **Amazon SES** | ✅ | ✅ (SMTP creds) | ❌ | ❌ | Requires TLS/STARTTLS |
| **Postmark** | ✅ | ✅ (server token) | ❌ | ❌ | Transactional focus |
| **Gmail SMTP** | ⚠️ (legacy only) | ❌ | ❌ | ✅ | Basic Auth disabled for most accounts |
| **Microsoft Exchange Online** | ❌ | ❌ | ❌ | ✅ | Modern Auth only |

\* “API key as password” rides over PLAIN/LOGIN - not a separate protocol.  

---

## Prioritization 
- **Must-have:**  
  - **PLAIN/LOGIN** → unlocks SendGrid, SES, Postmark (most common transactional providers).  
  - **XOAUTH2** → required for Gmail and Microsoft Exchange Online (without it, those integrations are impossible).  
  - **Custom SMTP hostname** → solves compatibility issues with strict servers that reject default hostnames.  
  - **Error handling & reset to defaults** → ensures users aren’t stuck with broken configs.  

- **Out of scope (for later):**  
  - CRAM-MD5 and other legacy SASL mechanisms (rare in SaaS).  
  - Provider-specific templates (nice DX, but not needed to unlock functionality).  

---

## Requirements (Functional, MVP Scope)  

| Requirement | User Functionality | Benefit |
|---|---|---|
| **Select SMTP auth type** | User can choose between PLAIN/LOGIN (username/password) and OAuth2 | Works with both modern (Microsoft/Gmail) and traditional (SendGrid, SES, Postmark) providers |
| **Configure provider-specific credentials** | User can enter API keys (SendGrid/Postmark), AWS SMTP creds (SES), or OAuth tokens (Microsoft/Gmail) | Reuses their existing provider setup with minimal friction |
| **Custom SMTP hostname** | User can set the hostname ZITADEL uses when introducing itself to SMTP servers | Fixes connection issues with strict providers that reject default hostnames |
| **Helpful error handling** | If config fails or no provider is set, user sees clear errors | Less frustration, fewer support tickets |

---

## More info from Stefan: 

Current configuration example https://zitadel.com/docs/apis/resources/admin/admin-service-update-smtp-config#update-smtp-configuration:
```
{
  "senderAddress": "noreply@m.zitadel.cloud",
  "senderName": "ZITADEL",
  "tls": true,
  "host": "smtp.postmarkapp.com:587",
  "user": "197f0117-529e-443d-bf6c-0292dd9a02b7",
  "replyToAddress": "replyto@m.zitadel.cloud",
  "password": "this-is-my-password",
  "description": "provider description"
}
```

Suggested changes:
- [ ] add boolean to configuration to add `hello` request
- [ ] add `oneof` for authtype, as Plain and Login are available already (user and password attributes)
- [ ] include CRAMMD5Auth as authtype (https://www.samlogic.net/articles/smtp-commands-reference-auth.htm)
- [ ] include EntraID as authtype (https://learn.microsoft.com/en-us/exchange/client-developer/legacy-protocols/how-to-authenticate-an-imap-pop-smtp-application-by-using-oauth#sasl-xoauth2, https://github.com/AzureAD/microsoft-authentication-library-for-go), is combined with CRAMMD5 (https://learn.microsoft.com/en-us/azure/azure-app-configuration/rest-api-authentication-azure-ad)
- [ ] endpoint to re-add default SMTP configuration
- [ ] reevaluate error handling if no SMTP provider is active and missing translations

Additional issues to take care of:
- [ ] https://github.com/zitadel/zitadel/issues/9361

**Comment by stebenz** [comment]
Closes #4477
Closes #5424 
Closes #7052 
Closes #7085 
Closes #7739

**Comment by hifabienne** [comment]
@eliobischof @stebenz please estimate

**Comment by doncicuto** [comment]
Hi @eliobischof , @stebenz would you find suitable to use https://github.com/wneessen/go-mail for some of these requirements?

**Comment by fireclubdevel** [comment]
I am currently having this very same problem with a postfix+dovecot setup. In other programs I found go-mail suitable. It is true that you add a dependency, but it works fine, even with attachments.

**Comment by JgdKdoFhr** [comment]
Another helpful feature would be to enable/disable authentication for SMTP completely. In my environment, I've set up my Exchange Online to accept Mails from several IP addresses, so I won't have to deal with authentication at all.

**Comment by elinashoko** [comment]
Adding more context here - Microsoft has turned off Basic Auth for most email protocols and will fully kill it for SMTP by March 2026. A lot of companies are already blocking it early because of security policies. So SMTP might already be breaking for some customers (although, I haven't seen anything reported). So we need to extend SMTP to support at least another oauth2 method and include this in v5, so that we can futureproof SMTP before March. 

What estimate did you, guys, land on? @livio-a @stebenz 

https://learn.microsoft.com/en-us/exchange/clients-and-mobile-in-exchange-online/deprecation-of-basic-authentication-exchange-online

**Comment by elinashoko** [comment]
**Notes**
- Mailgun SMTP provider also is popular (and also currently supported) 
- Clarifying: the main use case is transactional mails

**Comment by elinashoko** [comment]
To dos
- Console - will need to switch the endpoints, display the new templates that are needed
- Need to confirm the list of templates 
- Is outlook the same as Microsoft B2B mail? Confirm

**Comment by janwijbrand** [comment]
> Another helpful feature would be to enable/disable authentication for SMTP completely. In my environment, I've set up my Exchange Online to accept Mails from several IP addresses, so I won't have to deal with authentication at all.

Not sure how helpful it is for prioritization: we want do not require user/password either. This is currently not supported preventing us to go forward.
